src/Microsoft.PowerShell.ConsoleHost/host/msh/Telemetry.cs

```diff
@@ -1,102 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-using System;
-using System.Collections.Generic;
-using System.IO;
-using System.Management.Automation;
-using System.Reflection;
-using System.Runtime.InteropServices;
-using System.Security.Cryptography;
-
-using Microsoft.ApplicationInsights;
-using Microsoft.ApplicationInsights.DataContracts;
-using Microsoft.ApplicationInsights.Extensibility;
-
-namespace Microsoft.PowerShell
-{
-    /// <summary>
-    /// Send up telemetry for startup.
-    /// </summary>
-    internal static class ApplicationInsightsTelemetry
-    {
-        // If this env var is true, yes, or 1, telemetry will NOT be sent.
-        private const string TelemetryOptoutEnvVar = "POWERSHELL_TELEMETRY_OPTOUT";
-
-        // Telemetry client to be reused when we start sending more telemetry
-        private static TelemetryClient _telemetryClient = null;
-
-        // Set this to true to reduce the latency of sending the telemetry
-        private static bool _developerMode = false;
-
-        // PSCoreInsight2 telemetry key
-        private const string _psCoreTelemetryKey = "ee4b2115-d347-47b0-adb6-b19c2c763808";
-
-        static ApplicationInsightsTelemetry()
-        {
-            TelemetryConfiguration.Active.InstrumentationKey = _psCoreTelemetryKey;
-            TelemetryConfiguration.Active.TelemetryChannel.DeveloperMode = _developerMode;
```

Comment:
@JamesWTruher It turns out this two lines caused the following 3 assemblies to be loaded at the startup time.
In the new changes, you avoid running these 2 lines of code when the telemetry is turned off (_CI builds always turn off telemetry_), and hence the following 3 assemblies are not loaded at startup now.

This explains the failure to that startup test, but also reveals a problem -- that test is not tracking the real startup scenario, but only the telemetry-optout scenario.
Today, with the telemetry turned on, the telemetry code results in 6 assemblies being loaded at startup time.
I think we should run `startup.test.ps1` with telemetry turned on, but that will produce some noise to our telemetry data.

```
System.Private.Xml.Linq.dll
System.Threading.Timer.dll
System.Xml.XDocument.dll
```

```
Microsoft.ApplicationInsight.dll
System.Diagnostics.DiagnosticSource.dll
System.Net.NameResolution.dll
```